### PR TITLE
ffmpeg: add avisynth support

### PR DESCRIPTION
**Pull request by jopejoe1**
## Description of changes
depends on  #286292

Makes it possible for ffmpeg to read AviSynth scripts

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/av/avisynthplus/package.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    patches = [
+      (fetchpatch {
+        name = "fix-absolute-path.patch";
+        url = "https://github.com/AviSynth/AviSynthPlus/commit/818983691e962ec3e590fcad07032f8a139a6b16.patch";
+        hash = "sha256-4yUOnjtOroX+bhNUKbYz/giKaslzYdwPaaJWNkrTBr4=";
+      })
+    ];
+
+    buildInputs = [ cmake ];
```
Comment (AndersonTorres) [inline pkgs/by-name/av/avisynthplus/package.nix:30]:
```suggestion
    nativeBuildInputs = [ cmake ];
    
    strictDeps = true;
```

File: pkgs/by-name/av/avisynthplus/package.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  gitUpdater,
+  fetchpatch,
+}:
+
+stdenv.mkDerivation (
+  finalAttrs: {
```
Comment (AndersonTorres) [inline pkgs/by-name/av/avisynthplus/package.nix:11]: Was this because of formatter?
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
**Considering** that 
- avisynthplus is a new package
  - therefore it does not causes mass rebuilds
    - therefore, reviewing it in isolation is easier and cheaper 
      - especially for cheaper machines (like mine)
    - therefore it can be sent to master branch
      - making it readily available to be Hydra-cached

- ffmpeg causes massive rebuilds

**Split** the ffmpeg patch in a separate pull request.

File: pkgs/by-name/av/avisynthplus/package.nix (hunk 2 above)
Comment (jopejoe1) [reply pkgs/by-name/av/avisynthplus/package.nix:11]: yeah the rfc formatter did that.

**Comment by jopejoe1** [comment]
Move package creation to #286292

File: pkgs/by-name/av/avisynthplus/package.nix (hunk 1 above)
Comment (jopejoe1) [reply pkgs/by-name/av/avisynthplus/package.nix:30]: fixed that in #286292

**Review by arthsmn — APPROVED** [review]

**Comment by Atemu** [comment]
Do you have a sample script with which you tested this?

**Pull request merged**